packages/contracts/contracts/L1/ValidatorManager.sol

```diff
@@ -0,0 +1,497 @@
+     *
+     * @param validator Address of the validator.
+     *
+     * @return The status of the validator corresponding to the given address.
+     */
+    function getStatus(address validator) public view returns (ValidatorStatus) {
+        if (!_vaults[validator].isInitiated) {
+            return ValidatorStatus.NONE;
+        }
+
+        if (_jail[validator] != 0) {
+            return ValidatorStatus.IN_JAIL;
+        }
+
+        if (_vaults[validator].asset.validatorKro < MIN_REGISTER_AMOUNT) {
+            return ValidatorStatus.INACTIVE;
+        }
+
+        bool started = _validatorTree.nodeMap[validator] > 0;
+
+        // To prevent all MIN_START_AMOUNT is fulfilled with KRO in KGH which is not subject to slash,
+        // enable to start the validator when real asset satisfies the threshold.
+        if (
+            _reflectiveWeight(_vaults[validator]) - _vaults[validator].asset.totalKroInKgh <
+            MIN_START_AMOUNT
+        ) {
+            if (!started) {
+                return ValidatorStatus.ACTIVE;
+            }
+            return ValidatorStatus.STARTED;
+        }
+
+        if (!started) {
+            return ValidatorStatus.CAN_START;
+        }
+        return ValidatorStatus.CAN_SUBMIT_OUTPUT;
+    }
+
+    /**
+     * @notice Returns the no submission count of given validator.
+     *
+     * @param validator Address of the validator.
+     *
+     * @return The no submission count of given validator.
+     */
+    function noSubmissionCount(address validator) public view returns (uint8) {
+        return _vaults[validator].noSubmissionCount;
+    }
+
+    /**
+     * @notice Returns the total weight of started validators.
+     *
+     * @return The total weight of started validators.
+     */
+    function startedValidatorTotalWeight() public view returns (uint120) {
+        return _validatorTree.nodes[_validatorTree.root].weightSum;
+    }
+
+    /**
+     * @notice Returns the weight of given validator. It not started, returns 0.
+     *         Note that `weight / startedValidatorTotalWeight()` is the probability that the
+     *         validator is selected as a priority validator.
+     *
+     * @param validator Address of the validator.
+     *
+     * @return The weight of given validator.
+     */
+    function getWeight(address validator) public view returns (uint120) {
+        return _validatorTree.nodes[_validatorTree.nodeMap[validator]].weight;
+    }
+
+    /**
+     * @notice Updates next priority validator address. Validators with more delegation tokens have
+     *         a higher probability of being selected. The random weight selection is based on the
+     *         last finalized output root.
+     */
+    function _updatePriorityValidator() private {
+        uint120 weightSum = startedValidatorTotalWeight();
+        uint256 latestFinalizedOutputIndex = L2_ORACLE.latestFinalizedOutputIndex();
+
+        if (weightSum > 0 && latestFinalizedOutputIndex > 0) {
```

Comment:
Is there any reason for `latestFinalizedOutputIndex > 0` check? It seems that the `outputIndex` does not go to zero even after the `ValidatorPool` is terminated..